frontend/src/test/java/com/google/tripmeout/frontend/servlet/TripServletTest.java

```diff
@@ -0,0 +1,127 @@
+package com.google.tripmeout.frontend.servlet;
+
+import static com.google.common.truth.Truth.assertThat;
+import static org.mockito.Mockito.when;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.tripmeout.frontend.TripModel;
+import com.google.tripmeout.frontend.error.TripAlreadyExistsException;
+import com.google.tripmeout.frontend.error.TripNotFoundException;
+import com.google.tripmeout.frontend.serialization.GsonTripModelTypeAdapter;
+import com.google.tripmeout.frontend.servlet.TripServlet;
+import com.google.tripmeout.frontend.storage.InMemoryTripModelStorage;
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+public class TripServletTest {
+  private Gson gson;
+  private final static TripModel PARIS = TripModel.builder()
+                                             .setId("CDG")
+                                             .setName("Paris, France")
+                                             .setUserId("12345")
+                                             .setLocationLat(48.3288)
+                                             .setLocationLong(34.0)
+                                             .build();
+
+  @Mock HttpServletRequest request;
+
+  @Before
+  public void setUp() throws Exception {
+    MockitoAnnotations.initMocks(this);
+    this.gson = new GsonBuilder()
+                    .registerTypeAdapter(TripModel.class, new GsonTripModelTypeAdapter())
+                    .create();
+  }
+
+  @Test
+  public void returnsErrorCodeForBadIdForDoGet()
```

Comment:
Please rename test methods to reflect the method under test, the test case, and the expected behavior: `testMethod_testCase_behavior`. Or, if there's only one (obvious) behavior that describes the test case, just `testMethod_testCase`. Same applies to other test methods.

For example, you might call this case `doGet_badId_returnsNotFound`.